Catch2 (with CMake) does not compile when only test file discovered is empty

**Describe the bug**
When compiling in a project with no tests, but a (empty) tests file exists, the compile fails with the following error:
```
[ 50%] Linking CXX executable test
CMake Error at /usr/lib/cmake/Catch2/CatchAddTests.cmake:150 (string):
  string sub-command JSON expected an index less than 0 got '0'.
Call Stack (most recent call first):
  /usr/lib/cmake/Catch2/CatchAddTests.cmake:228 (catch_discover_tests_impl)
```

**Expected behavior**
The project should compile correctly.

**Reproduction steps**
Steps to reproduce the bug.
<!-- Usually this means a small and self-contained piece of code that uses Catch and specifying compiler flags if relevant. -->
1. Create base project directory (`base_dir`)
1. Add CMakeLists.txt ([CMakeLists.txt](https://github.com/user-attachments/files/19177887/CMakeLists.txt))
1. Create tests directory inside base (`base_dir/tests`)
1. Create an empty test file inside base (`base_dir/tests/tests.cpp`)
1. Create build directory inside base (`base_dir/build`)
1. `cd` into build directory
1. Run `cmake ..`
1. Run `make` -> an error occurs

**Platform information:**
<!-- Fill in any extra information that might be important for your issue. -->
 - OS: **Ubuntu 24.04.1 LTS arm64**
 - Compiler+version: **clang 18.1.3 (1ubuntu1)**
 - Catch version: **v3.8.0**


**Additional context**

--- a/extras/CatchAddTests.cmake
+++ b/extras/CatchAddTests.cmake
@@ -143,6 +143,12 @@ function(catch_discover_tests_impl)
   # Speed-up reparsing by cutting away unneeded parts of JSON.
   string(JSON test_listing GET "${listing_output}" "listings" "tests")
   string(JSON num_tests LENGTH "${test_listing}")
+
+  # Exit early if no tests are detected
+  if(num_tests STREQUAL "0")
+    return()
+  endif()
+
   # CMake's foreach-RANGE is inclusive, so we have to subtract 1
   math(EXPR num_tests "${num_tests} - 1")
 

